src/Views/IframeView.php

```diff
@@ -1,88 +1,289 @@
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setTitle( $title ) {
-		$this->title = $title;
+	public function setButtonTitle( $title ) {
+		$this->buttonTitle = $title;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Button color.
 	 *
-	 * @param string $body
+	 * @param string $color
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBody( $body ) {
-		$this->body = $body;
+	public function setButtonColor( $color ) {
+		$this->buttonColor = $color;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Form id.
 	 *
-	 * @param array $classes
+	 * @param int $id
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBodyClasses( $classes ) {
-		$this->bodyClasses = array_merge( $this->bodyClasses, (array) $classes );
+	public function setFormId( $id ) {
+		$this->formId = $id;
 
 		return $this;
 	}
 
+	/**
+	 * Get iframe HTML.
+	 *
+	 * @return string
+	 */
+	private function getIframeHTML() {
+		ob_start();
+		include $this->template->getLoadingView();
+
+		$loader = sprintf(
+			'<div class="iframe-loader">%1$s</div>',
+			ob_get_clean()
+		);
+
+		$iframe = sprintf(
+			'<iframe
+				name="give-embed-form"
+				%1$s
+				%4$s
+				data-autoScroll="%2$s"
+				onload="Give.initializeIframeResize(this)"
+				style="border: 0;visibility: hidden;%3$s"></iframe>%5$s',
+			$this->modal ? "data-src=\"{$this->url}\"" : "src=\"{$this->url}\"",
+			$this->autoScroll,
+			$this->minHeight ? "min-height: {$this->minHeight}px;" : '',
+			$this->modal ? 'class="in-modal"' : '',
+			$loader
+		);
+
+		if ( $this->modal ) {
+			$iframe = sprintf(
+				'<div class="modal-inner-wrap">
+					<div class="modal-content">
+		    			<a href="#" class="close-btn js-give-embed-form-modal-closer" aria-label="%3$s" data-form-id="%3$s" rel="nofollow">%2$s<span>&times;</span></a>
+						%1$s
+					</div>
+				</div>',
+				$iframe,
+				__( 'Close modal', 'give' ),
+				$this->uniqueId
+			);
+		}
+
+		return $iframe;
+	}
+
+
+	/**
+	 * Get button HTML.
+	 *
+	 * @return string
+	 */
+	private function getButtonHTML() {
+		return sprintf(
+			'<div class="js-give-embed-form-modal-launcher-wrap">
+				<button
+				type="button"
+				class="js-give-embed-form-modal-opener"
+				data-form-id="%1$s"%3$s>%2$s</button>
+			</div>',
+			$this->uniqueId,
+			$this->buttonTitle,
+			$this->buttonColor ? " style=\"background-color: {$this->buttonColor}\"" : ''
+		);
+	}
+
+	/**
+	 * Get iframe URL.
+	 *
+	 * @return string
+	 */
+	private function getIframeURL() {
+		$query_string           = array_map( 'give_clean', wp_parse_args( $_SERVER['QUERY_STRING'] ) );
+		$donation_history       = give_get_purchase_session();
+		$hasAction              = ! empty( $query_string['giveDonationAction'] );
+		$this->autoScroll       = absint( $hasAction );
+		$donationFormHasSession = $this->formId === absint( $donation_history['post_data'] ['give-form-id'] );
```

Comment:
Extra space between array keys

Reply:
@bradp formating autocorrect by the automated tool which we step on every `git commit` command. Since we are using WPCS, so I believe that most of the things formatting correctly.

ref: https://github.com/impress-org/givewp/pull/4659#discussion_r420883843